CONTRIBUTING.md

```diff
@@ -1,75 +1,389 @@
+
+Note that `assets` should store all the auxiliary files for the files in its parent folder. You don't make an `assets` folder
+or similar for individual files/pages.
+
+### README files
+
+VuePress converts Markdown files to same named HTML file. `README.md` files are an exception; they're 
+renamed `index.html`, since that's how web servers find a website's root file. Navigating 
+to `https://agoric.com/documentation/ertp/guide/` displays the VuePress processed `/main/ertp/guide/README.md`.
+While it may seem odd, VuePress expects multiple `README.md` files in a repo, one each for most folders in it.
+
+The root README.md file must start with a header. Any of H1, H2, or H3 (`#`, `##`, or `###` in Markdown) will do.
+This is needed to generate search indexes and sidebars.
+
+A subdirectory is invisible to VuePress unless it has a `README.md`. Empty `README.md` files are acceptable.
+
+Lines with no special treatment are converted into standard HTML paragraph tags.
+
+### Sectioning Pages
+
+VuePress automatically builds search functionality and individual page menus from `h1`, `h2`, and `h3` headers (i.e Markdown's `#`, `##`, and `###` commands). 
+You must have only **one** `h1` per `.md` file. Be careful not to have too many `h2` and `h3` level headers on one page.
+
+## Writing Links
+
+VuePress turns Markdown links into HTML links. There are some quirks about how you should write Markdown links.
+
+First, our link checker does **not** check `router-link` style links. Please only use Markdown style links.
+
+Next, your Markdown links should be to the `.md` Markdown files in the Doc repo. VuePress processing will change
+both the `.md` files and links to them to be `.html`.
+
+Use relative links instead of absolute ones for any links to files or folders in the Documentation repo. Relative links
+open in the same browser tab when clicked on, absolute links open a new tab. 
+
+Use this trick to make writing relative links easier. While it's easy enough to, say, write a relative link to something in
+the same folder as the file you're writing (something like `(./assets/my-diagram.svg)` to include an image), it can be
+tricky to remember/figure out what the right syntax is for linking to a file two folders up, in a different upper folder, and then
+two levels down from there on a different branch of the file structure. 
+
+Instead, `main` is considered the top of the file hierarchy. So you can always get to, say, a Glossary entry 
+by just linking to `(/glossary/#allocation)`; its path relative to `main`. Any path starting with just `/` is considered
+to start at `main`.
+
+VuePress turns every header in a Markdown file to a named file location you can link to so clicking take you directly to
+that file location. At the end of the link, append a `#` to the file name, followed by the header text. The header text
+must be altered to be 1) all lower case. 2) All punctuation and spaces are replaced by hyphens (except there aren't any
```

Comment:
Please correct this if I'm wrong about "punctuation and spaces" meaning "non-alphanumerics". I don't know what's included in "punctuation". Are multiple consecutive non-alphanumerics turned into a single dash or one-per-character?
```suggestion
must be altered to be 1) all lower case. 2) All non-alphanumerics are replaced by hyphens (except there aren't any
```

Reply:
That's a good question about multiple consecutive non-alphanumerics. I'll have to do some experimenting.